src/NLU.DevOps.Luis.Tests.Shared/RetryTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace NLU.DevOps.Luis.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using FluentAssertions;
+    using Moq;
+    using NUnit.Framework;
+
+    [TestFixture]
+    internal static class RetryTests
+    {
+        [Test]
+        public static void ParsesRetryAfterIntString()
+        {
+            var retryAfter = "42";
+            Retry.GetRetryAfterDelay(retryAfter, TimeSpan.Zero).TotalSeconds.Should().BeApproximately(42, 10e-6);
+        }
+
+        [Test]
+        public static void ParsesRetryAfterDateString()
+        {
+            var delta = 1;
+            var retryAfter = DateTimeOffset.Now.AddSeconds(10).ToString("r", CultureInfo.InvariantCulture);
+            var delay = Retry.GetRetryAfterDelay(retryAfter, TimeSpan.Zero);
+            delay.TotalSeconds.Should().BeApproximately(10, delta);
```

Comment:
I actually should not use `BeApproximately` here, because it's +/- delta. The problem with this test is that the implementation of `GetRetryAfterDelay` uses `DateTimeOffset.Now`, rather than some clock interface that can be easily mocked. So, this test is _technically_ non-deterministic because we don't know how much of a diff will be between our estimated clock time (on L31) and the clock time in the `GetRetryAfterDelay` implementation, though given this is a synchronous operation without much compute work, we can be pretty confident that it will be < O(100ms). Thus I chose 1 second as a possible diff value.

It's a good catch though, as this tests 9 <= delay <= 11 (due to `BeApproximately`) when what we really want is 9 <= delay <= 10.